src/components/ui/form/checkbox/CheckboxWithLabel.tsx

```diff
@@ -28,29 +28,32 @@ type InnerProps = CheckboxWithLabelProps &
 export const CheckboxWithLabelComponent: React.FC<InnerProps> = props => {
   const {
     children,
+    disabled,
     label,
+    onChange, // Do not pass to SimpleCheckbox
+    onToggle,
+    ref,
     textColor,
-    disabled,
     theme,
-    onToggle,
-    onChange, // Do not pass to SimpleCheckbox
     ...propsToCheckbox
   } = props;
   return (
-    <Clickable onClick={disabled ? undefined : onToggle}>
-      <Row alignItems={'center'}>
-        <SimpleCheckbox {...propsToCheckbox} disabled={disabled} />
-        <Space />
-        {label && (
-          <DefaultText
-            color={disabled ? theme.disabledColors.iconColor : textColor}
-          >
-            {label}
-          </DefaultText>
-        )}
-        {children}
-      </Row>
-    </Clickable>
+    <div ref={ref}>
+      <Clickable onClick={disabled ? undefined : onToggle}>
+        <Row alignItems={'center'}>
+          <SimpleCheckbox {...propsToCheckbox} disabled={disabled} />
```

Comment:
Maybe allow passing of refs to SimpleCheckbox?

Reply:
You can use input ref to be able to pass ref to the actual input.